src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java

```diff
@@ -68,6 +70,24 @@ public void testSetupChildFromBaseClass() throws CheckstyleException {
     }
 
     @Test
+    public void testSetupInvalidChildFromBaseClass() throws Exception {
+        final TestBean testBean = new TestBean();
+        final DefaultConfiguration conf = new DefaultConfiguration("testConf");
+        final Field field = AutomaticBean.class.getDeclaredField("configuration");
+        field.setAccessible(true);
+        field.set(testBean, conf);
+
+        try {
+            testBean.setupChild(conf);
+            fail("expecting checkstyle exception");
+        }
+        catch (CheckstyleException ex) {
+            assertEquals("expected exception", "testConf is not allowed as a child in testConf",
```

Comment:
@rnveach I think here we should have different name for parent and child to not confuse them